File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/card/GetCardUseCase.java
package com.lattis.ellipse.domain.interactor.card;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.model.Card;
import com.lattis.ellipse.domain.repository.CardRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 7/26/17.
 */

public class GetCardUseCase extends UseCase<List<Card>> {

    private CardRepository cardRepository;

    @Inject
    public GetCardUseCase(ThreadExecutor threadExecutor,
                          PostExecutionThread postExecutionThread,
                          CardRepository cardRepository) {
        super(threadExecutor, postExecutionThread);
        this.cardRepository = cardRepository;
    }

    @Override
    protected Observable<List<Card>> buildUseCaseObservable() {
        return cardRepository.getCard();
    }

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/card/GetCardUseCase.kt
package com.lattis.domain.usecase.card


import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.CardRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.Card
import javax.inject.Inject

import io.reactivex.rxjava3.core.Observable

/**
 * Created by ssd3 on 7/26/17.
 */

class GetCardUseCase @Inject
constructor(threadExecutor: ThreadExecutor,
            postExecutionThread: PostExecutionThread,
            private val cardRepository: CardRepository
) : UseCase<List<Card>>(threadExecutor, postExecutionThread) {

    override fun buildUseCaseObservable(): Observable<List<Card>> {
        return cardRepository.getCard()
    }

}
